python/lsst/ip/isr/isrStatistics.py

```diff
@@ -504,5 +554,171 @@ def copyCalibDistributionStatistics(self, inputExp, **kwargs):
+                                 np.arange(-self.config.biasShiftWindow + 1, 0)])
+        kernel = kernel/np.sum(kernel[:self.config.biasShiftWindow])
+
+        convolved = np.convolve(overscanData, kernel, mode='valid')
+        convolved = np.pad(convolved, (self.config.biasShiftWindow - 1, self.config.biasShiftWindow))
+
+        shift_check = np.abs(convolved)/noise
+        shift_mask = shift_check > self.config.biasShiftThreshold
+        shift_mask[:self.config.biasShiftSkip] = False
+
+        shift_regions = np.flatnonzero(np.diff(np.r_[np.int8(0),
+                                                     shift_mask.view(np.int8),
+                                                     np.int8(0)])).reshape(-1, 2)
+        shift_peaks = []
+        for region in shift_regions:
+            region_peak = np.argmax(shift_check[region[0]:region[1]]) + region[0]
+            if self._satisfies_flatness(region_peak, convolved[region_peak], overscanData):
+                shift_peaks.append(
+                    [float(convolved[region_peak]), float(region_peak),
+                     int(region[0]), int(region[1])])
+        return noise, shift_peaks
+
+    def _satisfies_flatness(self, shiftRow, shiftPeak, overscanData):
+        """
+        """
+        prerange = np.arange(shiftRow - self.config.biasShiftWindow, shiftRow)
+        postrange = np.arange(shiftRow, shiftRow + self.config.biasShiftWindow)
+
+        preFit = linregress(prerange, overscanData[prerange])
+        postFit = linregress(postrange, overscanData[postrange])
+
+        if shiftPeak > 0:
+            preTrend = (2*preFit[0]*len(prerange) < shiftPeak)
+            postTrend = (2*postFit[0]*len(postrange) < shiftPeak)
+        else:
+            preTrend = (2*preFit[0]*len(prerange) > shiftPeak)
+            postTrend = (2*postFit[0]*len(postrange) > shiftPeak)
+
+        return (preTrend and postTrend)
+
+    def measureAmpCorrelations(self, inputExp, overscanResults):
+        """Measure number of bias shifts from overscan data.
+
+        Parameters
+        ----------
+        inputExp : `lsst.afw.image.Exposure`
+            Exposure to measure.
+        overscans : `list` [`lsst.pipe.base.Struct`]
+            List of overscan results.  Expected fields are:
+
+            ``imageFit``
+                Value or fit subtracted from the amplifier image data
+                (scalar or `lsst.afw.image.Image`).
+            ``overscanFit``
+                Value or fit subtracted from the overscan image data
+                (scalar or `lsst.afw.image.Image`).
+            ``overscanImage``
+                Image of the overscan region with the overscan
+                correction applied (`lsst.afw.image.Image`). This
+                quantity is used to estimate the amplifier read noise
+                empirically.
+
+        Returns
+        -------
+        outputStats : `dict` [`str`, [`dict` [`str`,`float]]
```

Comment:
That is a complicated output!  Also missing a backtick.

Reply:
Fixed.